packages/router/router.ts

```diff
@@ -3094,7 +3094,7 @@ function normalizeTo(
   let path = resolveTo(
     to ? to : ".",
     getPathContributingMatches(contextualMatches).map((m) => m.pathnameBase),
-    location.pathname,
+    stripBasename(location.pathname, basename) || location.pathname,
```

Comment:
`useLocation` strips this off so this code brought in from the react layer lost that since the raw router location contains the basename